build/features.py

```diff
@@ -942,29 +943,34 @@ def description(self):
         return "Optimization and Tuning"
 
     def enabled(self, build):
-        build.flags['optimize'] = util.get_flags(build.env, 'optimize', 1)
-        if int(build.flags['optimize']):
+        build.flags['optimize'] = SCons.ARGUMENTS.get('optimize', '')
+        if build.flags['optimize'] is not 'portable':
             return True
         else:
             return False
 
     def add_options(self, build, vars):
-        if not build.platform_is_windows:
-            vars.Add(
-                'optimize', 'Set to:\n  1 for -O3 compiler optimizations\n  2 for Pentium 4 optimizations\n  3 for Intel Core optimizations\n  4 for Intel Core 2 optimizations\n  5 for Athlon-4/XP/MP optimizations\n  6 for K8/Opteron/AMD64 optimizations\n  7 for K8/Opteron/AMD64 w/ SSE3\n  8 for Celeron D (generic SSE/SSE2/SSE3) optimizations.', 1)
-        else:
-            if build.machine_is_64bit:
-                vars.Add(
-                    'optimize', 'Set to:\n  1 to maximize speed (/O2)\n  2 for maximum optimizations (/Ox)', 1)
-            else:
-                vars.Add(
-                    'optimize', 'Set to:\n  1 to maximize speed (/O2)\n  2 for maximum optimizations (/Ox)\n  3 to use SSE instructions\n  4 to use SSE2 instructions', 1)
+        vars.Add(
+            'optimize', 'Set to:\n' \
+                        '  portable (alias 9): sse2 CPU (>= Pentium 4)\n' \
+                        '  native: optimized exclusive for the CPU of this system\n' \
+                        '  legacy: pure i386 code' \
+                        '  disabled: no optimization, for debug only' \
+                        , 1)
 
     def configure(self, build, conf):
         if not self.enabled(build):
             return
 
-        optimize_level = int(build.flags['optimize'])
+        optimize_level = SCons.ARGUMENTS.get('optimize', 'portable')
+
+        if optimize_level == 'disabled':
+            self.status = "disabled: no optimazion, for debug only"
+            return
+
+        if not build.build_is_debug and not build.build_is_release and optimize_level == '':
```

Comment:
If one checks out mixxx and build with just 
scons 
he should get an SSE optimized debug build

This if branch is executed when the user has explicit forced no flags by build=none.
So I think I can remove the  < and optimize_level == ''" > part